presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/JtsGeometryUtils.java

```diff
@@ -0,0 +1,474 @@
+    private enum EsriShapeType
+    {
+        POINT(1),
+        POLYLINE(3),
+        POLYGON(5),
+        MULTI_POINT(8);
+
+        final int code;
+
+        EsriShapeType(int code)
+        {
+            this.code = code;
+        }
+
+        static EsriShapeType valueOf(int code)
+        {
+            for (EsriShapeType type : values()) {
+                if (type.code == code) {
+                    return type;
+                }
+            }
+
+            throw new IllegalArgumentException("Unsupported ESRI shape code: " + code);
+        }
+    }
+
+    /**
+     * Shape type defined by JTS.
+     */
+    private static final String JTS_POINT = "Point";
+    private static final String JTS_POLYGON = "Polygon";
+    private static final String JTS_LINESTRING = "LineString";
+    private static final String JTS_MULTI_POINT = "MultiPoint";
+    private static final String JTS_MULTI_POLYGON = "MultiPolygon";
+    private static final String JTS_MULTI_LINESTRING = "MultiLineString";
+    private static final String JTS_GEOMETRY_COLLECTION = "GeometryCollection";
+
+    private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory();
+
+    private JtsGeometryUtils() {}
+
+    /**
+     * Deserializes ESRI shape as described in
+     * https://www.esri.com/library/whitepapers/pdfs/shapefile.pdf
+     * into JTS Geometry object.
+     */
+    public static Geometry deserialize(Slice shape)
+    {
+        if (shape == null) {
+            return null;
+        }
+
+        BasicSliceInput input = shape.getInput();
+        input.skip(SIZE_OF_INT); // SRID
+
+        if (input.available() == 0) {
+            return GEOMETRY_FACTORY.createGeometryCollection();
+        }
+
+        // GeometryCollection: spatialReferenceId|len-of-shape1|bytes-of-shape1|len-of-shape2|bytes-of-shape2...
+        List<Geometry> geometries = new ArrayList<>();
+        while (input.available() > 0) {
+            int length = input.readInt();
+            geometries.add(readGeometry(input.readSlice(length).getInput()));
+        }
+
+        if (geometries.size() == 1) {
+            return geometries.get(0);
+        }
+
+        return GEOMETRY_FACTORY.createGeometryCollection(geometries.toArray(new Geometry[0]));
+    }
+
+    /**
+     * Serialize JTS {@link Geometry} shape into an ESRI shape
+     */
+    public static Slice serialize(Geometry geometry) {
+        int spatialReferenceId = geometry.getSRID();
+        SliceOutput output = new DynamicSliceOutput(100);
+        output.appendInt(spatialReferenceId);
+
+        if (geometry.getGeometryType().equals(JTS_GEOMETRY_COLLECTION)) {
+            for (int i = 0; i < geometry.getNumGeometries(); i++) {
+                appendGeometry(geometry.getGeometryN(i), output);
+            }
+        } else {
+            appendGeometry(geometry, output);
+        }
+        return output.slice();
+    }
+
+    private static void appendGeometry(Geometry geometry, SliceOutput output) {
```

Comment:
s/appendGeometry/writeGeometry for consistency with other methods (e.g. writePoint, writePolygon, etc.)